routers/web/repo/editor.go

```diff
@@ -420,10 +420,10 @@ func DiffPreviewPost(ctx *context.Context) {
 	}
 
 	if diff.NumFiles == 0 {
-		ctx.HTMLString(http.StatusOK, `<div class="tw-p-6">`+ctx.Locale.TrString("repo.editor.no_changes_to_show")+`</div>`)
-		return
+		ctx.Data["File"] = nil
+	} else {
+		ctx.Data["File"] = diff.Files[0]
```

Comment:
```
if diff.NumFiles != 0 {
    ctx.Data["File"] = diff.Files[0]
}
```

Reply:
Is it safe from out-of-bounds access?

Reply:
Ah yes, template accessing `nil` should be fine.

Reply:
Where is the ` out-of-bounds `? The logic is the same as before.

Reply:
It's fine, I misread.